BuildTasks/PublishExtension/task.json

```diff
@@ -1,210 +1,231 @@
+    {
+      "name": "extensionId",
+      "type": "string",
+      "label": "Extension ID",
+      "defaultValue": "",
+      "helpMarkDown": "Overrides extension ID. If not specified, the extension ID specified in the manifest will be used",
+      "required": false,
+      "groupName": "manifest"
+    },
+    {
+      "name": "extensionTag",
+      "type": "string",
+      "label": "Extension Tag",
+      "defaultValue": "",
+      "helpMarkDown": "Extension Tag to append to the extension ID",
+      "required": false,
+      "groupName": "manifest"
+    },
+    {
+      "name": "extensionName",
+      "type": "string",
+      "label": "Extension name",
+      "defaultValue": "",
+      "helpMarkDown": "Overrides extension name. If not specified, the extension name specified in the manifest will be used",
+      "required": false,
+      "groupName": "manifest"
+    },
+    {
+      "name": "extensionVersion",
+      "type": "string",
+      "label": "Extension version",
+      "defaultValue": "",
+      "helpMarkDown": "Overrides extension version. If not specified, the extension version specified in the manifest will be used",
+      "required": false,
+      "groupName": "manifest"
+    },
+    {
+      "name": "updateTasksVersion",
+      "type": "boolean",
+      "label": "Override tasks version",
+      "defaultValue": "true",
+      "required": false,
+      "helpMarkDown": "Search for contributed tasks in extension manifests and updates the version specified in each Build and Release task found.",
+      "groupName": "manifest"
+    },
+    {
+      "name": "extensionVisibility",
+      "type": "pickList",
+      "label": "Extension visibility",
+      "defaultValue": "default",
+      "helpMarkDown": "Overrides extension visibility (Public vs Private) and optionally adds the Preview flag. If not specified, the extension visibility specified in the manifest will be used",
+      "required": false,
+      "options": {
+        "default": "Not set",
+        "private": "Private",
+        "privatepreview": "Private Preview",
+        "publicpreview": "Public Preview",
+        "public": "Public"
+      },
+      "groupName": "manifest"
+    },
+    {
+      "name": "outputVariable",
+      "type": "string",
+      "label": "Output Variable",
+      "defaultValue": "Extension.OutputPath",
+      "required": false,
+      "helpMarkDown": "The variable name to assign the location of the generated package to.",
+      "groupName": "manifest",
+      "visibleRule": "fileType = manifest"
+    },
+    {
+      "name": "shareWith",
+      "type": "string",
+      "label": "Share with",
+      "defaultValue": "",
+      "helpMarkDown": "Comma separated list of accounts with which to share the extension if it's private (e.g. account_x,account_y,account_z)",
+      "required": false,
+      "groupName": "manifest",
+      "visibleRule": "extensionVisibility = private || extensionVisibility = privatepreview || extensionVisibility = default"

```

Comment:
Not possible, it's in another group, so you can't reference it from the VisibleRule